### ISSUE TITLE
Pre-Alpha testing thread

### ISSUE THREAD

**Issue by FleetAdmiralButter** [marzent/XIV-on-Mac#3]
Thread for anyone helping us test out the pre-alpha app.

When posting general feedback, please try and use the following format:

Mac model:
CPU/GPU/RAM:
Overall comments:
Performance:

If you're reporting a specific bug or issue, please try and use this format:

Mac model:
CPU/GPU/RAM:
Screenshot/Video:
Steps to reproduce:
Expected Results:
Actual Results:
Type: Graphical Issue/Install Issue/Crash or Hang/Gameplay/HID Issue/Bad Performance
Logs (if available):

**Comment by seathasky** [comment]
Mac model: Hackintosh
CPU/GPU/RAM: i7 10700k / 5700 XT / 32gb DDR4 
Screenshot/Video:
![Screen Shot 2021-12-27 at 3 07 32 AM](https://user-images.githubusercontent.com/17598024/147450109-03fab3c5-dbf7-4e0f-9ae0-2317e73da557.png)


Steps to reproduce: Clean prefix in ~/Library/Application Support/XIV on Mac folder
Expected Results: Dependencies to install
Actual Results: Application stops responding.
Type: App wont get past dependencies on fresh prefix.
Logs:
[wine.log](https://github.com/marzent/XIV-on-Mac/files/7779134/wine.log)

TLDR: Testing on notarized 1.0 alpha. App seemed to work on existing prefix but not on fresh prefix.

**Comment by marzent** [comment]
> TLDR: Testing on notarized 1.0 alpha. App seemed to work on existing prefix but not on fresh prefix.

This is very curious in that it does not want to load the supplied .dylibs as fallback, especially since it works on an existing prefix...
Can't reproduce the error unfortunately  :(
Anyways this is not an issue with the installer itself but a wine <-> dyld interaction, can you test with an earlier un-notarised build and see if the issues go away?

**Comment by seathasky** [comment]
> > TLDR: Testing on notarized 1.0 alpha. App seemed to work on existing prefix but not on fresh prefix.
> 
> This is very curious in that it does not want to load the supplied .dylibs as fallback, especially since it works on an existing prefix... Can't reproduce the error unfortunately :( Anyways this is not an issue with the installer itself but a wine <-> dyld interaction, can you test with an earlier un-notarised build and see if the issues go away?

Testing on latest build prior to github alpha on fresh prefix:
![Screen Shot 2021-12-27 at 3 27 58 AM](https://user-images.githubusercontent.com/17598024/147451914-7f0e7c2b-6a0b-4068-ab53-9d665f3a42e1.png)
![Screen Shot 2021-12-27 at 3 30 59 AM](https://user-images.githubusercontent.com/17598024/147452172-2116f101-6bc7-4417-8518-477148493ff4.png)

No issues installing dependencies on fresh prefix.

**Comment by seathasky** [comment]
Copying "Wine" libs from previous build to latest fixes dependency install as suggested @marzent

**Comment by Gcenx** [comment]
> Copying "Wine" libs from previous build to latest 1.0 alpha fixes dependency install as suggested @marzent
> 
> ![Screen Shot 2021-12-27 at 3 36 55 AM](https://user-images.githubusercontent.com/17598024/147452994-2361fc6f-05b3-416d-9320-5d767dc02f8b.png)

Was this from a prior version with a lot more dylibs (Wineskins runtime)?

**Comment by kloostec** [comment]
I'm having a very similar issue to seathasky, but with different hardware:

Mac model: M1 Macbook Air
CPU/GPU/RAM: M1 8 GPU/8 GPU, 16GB RAM
Screenshot/Video: Same behaviour as seathasky above
Steps to reproduce: Nuke ~Library/Application Support/XIV on Mac, then run the notarized build
Expected Results: Dependencies install correctly
Actual Results: Hang on dotNetFx40_Full_x86_64
Type: Install Issue
Logs (if available): (attached)

More notes:

* Force closing the wine32on64 process will start the next install, which also hangs.  This continues all the way until the xivonmac gui launches successfully, but clicking on play doesn't work of course
[wine.log](https://github.com/marzent/XIV-on-Mac/files/7781544/wine.log)
* I have never run the crosstie version and don't have Crossover installed.  I do have a Mac license for FFXIV and run the official launcher on this machine.
* I don't seem to have access to the non-notarized build so can't test that option to see if it works
[wine.log](https://github.com/marzent/XIV-on-Mac/files/7781553/wine.log)
.

**Comment by seathasky** [comment]
> I'm having a very similar issue to seathasky, but with different hardware:

This is fixed now in our next build. 👍🏽 

![Screen Shot 2021-12-27 at 3 10 02 PM](https://user-images.githubusercontent.com/17598024/147504102-22e4e3a0-4365-4cba-a038-7a0727cca045.png)

**Comment by marzent** [comment]
> > Copying "Wine" libs from previous build to latest 1.0 alpha fixes dependency install as suggested @marzent
> > ![Screen Shot 2021-12-27 at 3 36 55 AM](https://user-images.githubusercontent.com/17598024/147452994-2361fc6f-05b3-416d-9320-5d767dc02f8b.png)
> 
> Was this from a prior version with a lot more dylibs (Wineskins runtime)?

These were exactly the same dylibs except that they were notarized correctly and funnily https://github.com/marzent/XIV-on-Mac/blob/3709fdd399fa06df173f92e6af729a8b9a960c1c/XIV%20on%20Mac/Util.swift#L167 fixes it…
This in conjunction with it miraculously loading all libraries when the prefix already exists probably means that it must be some sort of `dyld` bug (also apparently it worked once on a Big Sur machine but then never again)

**Comment by Gcenx** [comment]
> These were exactly the same dylibs except that they were notarized correctly and funnily
> 
> https://github.com/marzent/XIV-on-Mac/blob/3709fdd399fa06df173f92e6af729a8b9a960c1c/XIV%20on%20Mac/Util.swift#L167
> 
> fixes it…

This would be the first project that uses this env option as CrossOver/Wineskin/PlayOnMac all only use `DYLD_FALLBACK_LIBRARY_PATH`, none of these use libraries provided via brew so this could be a factor.

> This in conjunction with it miraculously loading all libraries when the prefix already exists probably means that it must be some sort of `dyld` bug (also apparently it worked once on a Big Sur machine but then never again)

The issue is lightly partly related to [bug 49199](https://bugs.winehq.org/show_bug.cgi?id=49199) and bundling libraries built using a newer SDK and running them on an older version of macOS.

If the project is to target say macOS 10.15?, then all libraries should be compiled on that version of macOS or at least using the relevant MacOSX SDK and force the desired deployment target.

**Comment by xazander** [comment]
When posting general feedback, please try and use the following format:

Mac model: M1 Macbook Air 
CPU/GPU/RAM: 8cpu / 8gpu
Overall comments: Installing the dependencies took time to install but eventually after waiting 20-30mins, it was able to work.
Performance: 60fps normally and drops to 45fps at times which is really good

The best thing about this is that it lets me use my Windows license while on a Mac.  I haven't upgraded my license on the Mac from ShB so I was surprised to see it unlock Endwalker.

My experience was very manual where I ended up installing the dependencies manually 'coz I initially had an anti-virus installed.  After removing that, it was pretty much smooth sailing.

**Comment by Ceolona** [comment]
General feedback...

MBA M1 - 4/4/8 core
"High" preset (middle setting), changes to 1680x1050, full screen, occlusion culling and glare off
35 to 60+ PFS, depending on area.  (Limsa plaza the worst, as expected)

No significant complaints.

**Comment by evanlouie** [comment]
General feedback:

It would be great if the "Transmission in progress..." message was able to be minimized or or at the very least not fixed in the center of the screen during initial install

<img width="345" alt="Screen Shot 2021-12-30 at 3 20 22 PM" src="https://user-images.githubusercontent.com/2081587/147794125-f5ff9772-6198-40b1-8d43-c5fcf6f1d81f.png">

**Comment by seathasky** [comment]
> General feedback:
> 
> It would be great if the "Transmission in progress..." message was able to be minimized or or at the very least not fixed in the center of the screen during initial install
> 

Unfortunately this is our of our control. We are not the developers of XIVLauncher, please use their issue tracker: https://github.com/goatcorp/FFXIVQuickLauncher

**Comment by zhang-chengyu** [comment]
MacBook Pro 13 M1 8/8/16GB running Monterey 12.1, XIV on Mac version 1.1.0

Is it normal that every time I open app, it goes through the whole dependency installation process again? Not sure if I am doing the right thing since I don't know if I can directly open XIVLaucher.

Also, XIVLaucher crashed multiple times during downloading the game (twice simply crashed without warning, once completely frozen, once crashed with some wine warning message that I forgot to save). Not sure if this is the right place to report.

Other than the above issues, no complaint at all for the performance. Full screen at the resolution slightly higher than 1080P (1152P I think), High preset with shadow resolution changed to high, real-time reflection standard, grass quality high, 8x anisotropic filtering, low-detailed model on distant objects off, and low-detailed model on shadows off. 35-40FPS at Crystarium Aetheryte Plaza, very smooth in dungeons and trials (didn't check fps).

**Comment by seathasky** [comment]
> MacBook Pro 13 M1 8/8/16GB running Monterey 12.1, XIV on Mac version 1.1.0
> 
> Is it normal that every time I open app, it goes through the whole dependency installation process again? Not sure if I am doing the right thing since I don't know if I can directly open XIVLaucher.
> 
> Also, XIVLaucher crashed multiple times during downloading the game (twice simply crashed without warning, once completely frozen, once crashed with some wine warning message that I forgot to save). Not sure if this is the right place to report.
> 
> Other than the above issues, no complaint at all for the performance. Full screen at the resolution slightly higher than 1080P (1152P I think), High preset with shadow resolution changed to high, real-time reflection standard, grass quality high, 8x anisotropic filtering, low-detailed model on distant objects off, and low-detailed model on shadows off. 35-40FPS at Crystarium Aetheryte Plaza, very smooth in dungeons and trials (didn't check fps).

The dependancy thing is a bug and will be fixed in next release

**Comment by xis666** [comment]
The black screen of character selection is always loading

<img width="1512" alt="截屏2022-01-01 下午9 10 53" src="https://user-images.githubusercontent.com/58455001/147851583-6b229253-4e21-4a9a-ba7a-0690d463edd0.png">

**Comment by seathasky** [comment]
> The black screen of character selection is always loading
> 

I replied to your issue already, please go check

**Comment by xis666** [comment]
Mac model:
CPU/GPU/RAM: M1max32 / 64G / monterey 12.1
Graphical Issue：The external monitor enters the black screen of the game, and the controller cannot be used。

https://user-images.githubusercontent.com/58455001/148060811-a43caee8-bffb-4da5-9803-eae5c97242ee.mp4

**Comment by FleetAdmiralButter** [comment]
Hi @xis666, that happens when you run the app in Exclusive Fullscreen. Please change it to Borderless Windowed.

**Comment by xis666** [comment]
hi @FleetAdmiralButter ，Windowed games are double-displayed, and the picture is particularly poor. Why don't the windows double-display?

**Comment by marzent** [comment]
Closing since we are in alpha now :)

New issues can be filed separately
